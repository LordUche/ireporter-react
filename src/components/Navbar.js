import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import {
  Menu,
  Image,
  Header,
  Container,
  Button,
  Responsive,
  Icon,
  Segment,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import brand from '../../public/images/brand.svg';
import { logoutUser } from '../redux/actions/auth';

export const Navbar = ({ loggedIn, openSidebar, logout, user }) => {
  return (
    <nav className="navbar">
      <Segment inverted>
        <Menu inverted secondary>
          <Container>
            <Menu.Item icon as={Link} to="/" className="navbar__brand">
              <Image
                className="navbar__brand--icon"
                src={brand}
                size="mini"
                spaced="right"
              />
              <Header className="navbar__brand--text">iREPORTER</Header>
            </Menu.Item>
            <Responsive
              minWidth={Responsive.onlyTablet.minWidth + 1}
              as={Menu.Menu}
              position="right"
            >
              {!loggedIn && (
                <Fragment>
                  <Menu.Item as={Link} to="/signup" name="sign up" />
                  <Menu.Item as={Link} to="/login" name="log in" />
                </Fragment>
              )}
              {loggedIn && (
                <Fragment>
                  <Menu.Item as={Link} color="blue" to="/profile">
                    <Icon name="user circle" size="large" />
                    <span>{user ? user.username : ''}</span>
                  </Menu.Item>
                  <Menu.Item name="logout" as={Link} to="/">
                    <Button
                      color="red"
                      onClick={logout}
                      basic
                      content="Logout"
                    />
                  </Menu.Item>
                </Fragment>
              )}
            </Responsive>
            <Responsive
              maxWidth={Responsive.onlyTablet.minWidth}
              as={Menu.Menu}
              position="right"
            >
              <Menu.Item
                as={Button}
                icon="bars"
                className="navbar__humbugger"
                onClick={openSidebar}
              />
            </Responsive>
          </Container>
        </Menu>
      </Segment>
    </nav>
  );
};

Navbar.defaultProps = { user: null };

Navbar.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  openSidebar: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  user: PropTypes.shape({ username: PropTypes.string }),
};

const mapStateToProps = state => ({ user: state.auth.user });

export default connect(
  mapStateToProps,
  { logout: logoutUser }
)(Navbar);
