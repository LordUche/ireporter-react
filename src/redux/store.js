import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { logger } from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const middleware = [thunk];

process.env.NODE_ENV === `development` && middleware.push(logger);

const initialState = { auth: { loggedIn: false, loading: false } };
const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware))
);

export default store;
