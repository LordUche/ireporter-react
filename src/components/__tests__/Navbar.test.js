import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import Navbar from '../Navbar';
import store from '../../redux/store';
import mockRouterOptions from '../../../test/__mocks__/mockRouter';

describe('<Navbar />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <Navbar loggedIn={false} logout={jest.fn()} openSidebar={jest.fn()} />
        </Router>
      </Provider>,
      mockRouterOptions
    );
  });

  it('should match snapshot', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
