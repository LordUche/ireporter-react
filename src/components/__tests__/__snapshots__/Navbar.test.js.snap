// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "iupbkx",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "iupbkx",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ConnectFunction
        loggedIn={false}
        logout={[MockFunction]}
        openSidebar={[MockFunction]}
      >
        <Navbar
          loggedIn={false}
          logout={[Function]}
          openSidebar={[MockFunction]}
          user={null}
        >
          <nav
            className="navbar"
          >
            <Segment
              inverted={true}
            >
              <div
                className="ui inverted segment"
              >
                <Menu
                  inverted={true}
                  secondary={true}
                >
                  <div
                    className="ui inverted secondary menu"
                  >
                    <Container>
                      <div
                        className="ui container"
                      >
                        <MenuItem
                          as={[Function]}
                          className="navbar__brand"
                          icon={true}
                          to="/"
                        >
                          <Link
                            className="icon item navbar__brand"
                            onClick={[Function]}
                            to="/"
                          >
                            <a
                              className="icon item navbar__brand"
                              href="/"
                              onClick={[Function]}
                            >
                              <Image
                                as="img"
                                className="navbar__brand--icon"
                                size="mini"
                                spaced="right"
                                src="test-file-stub"
                                ui={true}
                              >
                                <img
                                  className="ui mini right spaced image navbar__brand--icon"
                                  src="test-file-stub"
                                />
                              </Image>
                              <Header
                                className="navbar__brand--text"
                              >
                                <div
                                  className="ui header navbar__brand--text"
                                >
                                  iREPORTER
                                </div>
                              </Header>
                            </a>
                          </Link>
                        </MenuItem>
                        <Responsive
                          as={[Function]}
                          getWidth={[Function]}
                          minWidth={769}
                          position="right"
                        >
                          <MenuMenu
                            position="right"
                          >
                            <div
                              className="right menu"
                            >
                              <MenuItem
                                as={[Function]}
                                name="sign up"
                                to="/signup"
                              >
                                <Link
                                  className="item"
                                  onClick={[Function]}
                                  to="/signup"
                                >
                                  <a
                                    className="item"
                                    href="/signup"
                                    onClick={[Function]}
                                  >
                                    Sign Up
                                  </a>
                                </Link>
                              </MenuItem>
                              <MenuItem
                                as={[Function]}
                                name="log in"
                                to="/login"
                              >
                                <Link
                                  className="item"
                                  onClick={[Function]}
                                  to="/login"
                                >
                                  <a
                                    className="item"
                                    href="/login"
                                    onClick={[Function]}
                                  >
                                    Log In
                                  </a>
                                </Link>
                              </MenuItem>
                            </div>
                          </MenuMenu>
                        </Responsive>
                        <Responsive
                          as={[Function]}
                          getWidth={[Function]}
                          maxWidth={768}
                          position="right"
                        />
                      </div>
                    </Container>
                  </div>
                </Menu>
              </div>
            </Segment>
          </nav>
        </Navbar>
      </ConnectFunction>
    </Router>
  </MemoryRouter>
</Provider>
`;
